org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/BuildPathManager.java

```diff
@@ -666,7 +666,7 @@ private boolean requiresUpdate(IProject p, IProgressMonitor monitor) {
             if(moduleDescription == null) {
               return false;
             }
-            String location = p.getLocation().toString();
+            String location = p.getLocationURI().toString();
```

Comment:
Couldn't we store the URI directly?

Reply:
Yes, I agree that using URIs in the map would be more naturally. I'll update the PR towards that end.

Reply:
One remark here: URI defines a rather complex equals/hash code contract (what might be irrelevant here) just something to take in mind that two URIs are considered equal even if there String representation is NOT equal!